### PR TITLE
Feature/1297/264 TP DTO timestamps

### PR DESCRIPTION
**Pull request by algattik**
## What this PR changes/adds

Add timestamps when querying TransferProcesses in Data management API.

## Why it does that

Useful for web front-ends, to allow to sort transfers by dates.

## Linked Issue(s)

#264

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: spi/transfer-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/transfer/TransferProcess.java [hunk 1]
```diff
@@ -100,6 +100,7 @@ public class TransferProcess implements TraceCarrier {
 
     private String id;
     private Type type = Type.CONSUMER;
+    private long createdTimestamp;
```
Comment (chrislomonico) [inline spi/transfer-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/transfer/TransferProcess.java:103]: nit: align member declaration of `createdTimestamp` to declaration of stateTimestamp

**Review by chrislomonico — APPROVED** [review]
only one small nit

File: spi/transfer-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/transfer/TransferProcess.java (hunk 1 above)
Comment (algattik) [reply spi/transfer-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/transfer/TransferProcess.java:103]: Doesn't it make more sense as now to have all `state*` fields together?

```java
    private long createdTimestamp;
    private int state;
    private int stateCount = UNSAVED.code();
    private long stateTimestamp;
```

File: extensions/sql/transfer-process-store-sql/src/main/java/org/eclipse/dataspaceconnector/sql/transferprocess/store/SqlTransferProcessStore.java [hunk 2]
```diff
@@ -53,6 +54,7 @@ public class SqlTransferProcessStore implements TransferProcessStore {
     private final TransferProcessStoreStatements statements;
     private final String leaseHolderName;
     private final SqlLeaseContextBuilder leaseContext;
+    protected Clock clock = Clock.systemUTC();
```
Comment (cpeeyush) [inline extensions/sql/transfer-process-store-sql/src/main/java/org/eclipse/dataspaceconnector/sql/transferprocess/store/SqlTransferProcessStore.java:57]: nit: unused.
```suggestion
```

File: extensions/sql/transfer-process-store-sql/src/main/java/org/eclipse/dataspaceconnector/sql/transferprocess/store/SqlTransferProcessStore.java [hunk 3]
```diff
@@ -36,6 +36,7 @@
 import java.sql.Connection;
 import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.time.Clock;
```
Comment (cpeeyush) [inline extensions/sql/transfer-process-store-sql/src/main/java/org/eclipse/dataspaceconnector/sql/transferprocess/store/SqlTransferProcessStore.java:39]: nit: unused.
```suggestion
```

File: extensions/sql/transfer-process-store-sql/src/main/java/org/eclipse/dataspaceconnector/sql/transferprocess/store/TransferProcessStoreStatements.java [hunk 4]
```diff
@@ -74,6 +74,10 @@ default String getTypeColumn() {
         return "type";
     }
 
+        default String getCreatedTimestampColumn() {
```
Comment (cpeeyush) [inline extensions/sql/transfer-process-store-sql/src/main/java/org/eclipse/dataspaceconnector/sql/transferprocess/store/TransferProcessStoreStatements.java:77]: nit: incorrect indentation.

**Pull request merged**